### PR TITLE
[+] Feature "read external endpoint configuration" has been implemented

### PR DESCRIPTION
**Pull request by Vladislav-Fedorenko**
**What was done:**
  * interface for external endpoint configuration reader has been added to Endpoint feature
  * implementation for reading external endpoint configuration has been added to Endpoint feature
  * plugin of ExternalEndpointConfigurationFileReader has been added to Endpoint-plugins feature
  * functional for reading external endpoint configuration has been added to HttpEndpointPlugin and HttpsEndpointPlugin
  * version of the feature Endpoint has been increased to 0.6.2
  * version of the feature Endpoint-plugins has been increased to 0.6.1
  * version of the feature HttpEndpoint-plugins has been increased to 0.6.4
  * version of the feature HttpsEndpoint-plugins has been increased to 0.6.4

These changes allow to apply external endpoint configuration.
Currently only the reading the configuration from file is implemented.
For using it you should add these fields to endpoint object:
```
{
  ... // default configuration
  "externalConfigurationReader": "endpoint external configuration file reader",
  "pathToExternalConfiguration": "path/to/properties.file"
}
```
In that case default configuration will be overridden values from external file.

### CODE REVIEW

**Comment by Vladislav-Fedorenko** [comment]
@akutalev please, review this PR

File: CoreFeatures/Endpoint/EndpointExternalConfigurationFileReader/src/main/java/info/smart_tools/smartactors/endpoint/endpoint_external_configuration_file_reader/EndpointExternalConfigurationFileReader.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     * This constructor loads external configuration from file that lays in configuration.pathToExternalConfiguration
+     * @param configuration                                 default endpoint configuration
+     * @throws EndpointExternalConfigurationReaderException if something went wrong while loading
+     */
+    public EndpointExternalConfigurationFileReader(final IObject configuration) throws EndpointExternalConfigurationReaderException {
+        try {
+            IKey fieldNameKey = Keys.getKeyByName(IFieldName.class.getCanonicalName());
+            IFieldName pathToExternalConfigurationFN = IOC.resolve(fieldNameKey, "pathToExternalConfiguration");
+
+            FileReader propsReader = new FileReader((String) configuration.getValue(pathToExternalConfigurationFN));
```
Comment (akutalev) [inline CoreFeatures/Endpoint/EndpointExternalConfigurationFileReader/src/main/java/info/smart_tools/smartactors/endpoint/endpoint_external_configuration_file_reader/EndpointExternalConfigurationFileReader.java:35]: I see FileReader creation, but i don't see call to method close()

File: CoreFeatures/EndpointServiceStarter/EndpointStarter/src/main/java/info/smart_tools/smartactors/endpoint_service_starter/endpoint_starter/EndpointsSectionProcessingStrategy.java [hunk 2]
```diff
@@ -43,6 +43,7 @@
  *                     "secure": true,
  *                     "httpOnly": true
  *                 }
```
Comment (akutalev) [inline CoreFeatures/EndpointServiceStarter/EndpointStarter/src/main/java/info/smart_tools/smartactors/endpoint_service_starter/endpoint_starter/EndpointsSectionProcessingStrategy.java:45]: missed comma

**Review by akutalev — CHANGES_REQUESTED** [review]

File: CoreFeatures/EndpointServiceStarter/EndpointStarter/src/main/java/info/smart_tools/smartactors/endpoint_service_starter/endpoint_starter/EndpointsSectionProcessingStrategy.java (hunk 2 above)
Comment (Vladislav-Fedorenko) [reply CoreFeatures/EndpointServiceStarter/EndpointStarter/src/main/java/info/smart_tools/smartactors/endpoint_service_starter/endpoint_starter/EndpointsSectionProcessingStrategy.java]: Fixed

File: CoreFeatures/Endpoint/EndpointExternalConfigurationFileReader/src/main/java/info/smart_tools/smartactors/endpoint/endpoint_external_configuration_file_reader/EndpointExternalConfigurationFileReader.java (hunk 1 above)
Comment (Vladislav-Fedorenko) [reply CoreFeatures/Endpoint/EndpointExternalConfigurationFileReader/src/main/java/info/smart_tools/smartactors/endpoint/endpoint_external_configuration_file_reader/EndpointExternalConfigurationFileReader.java]: Fixed. I moved creating FileReader to the resources of try-catch block

**Review by akutalev — APPROVED** [review]

**Comment by Colbys** [comment]
Merged manually for 0.7.0, closing

**Pull request closed without merging**